index.html

```diff
@@ -1884,50 +1909,72 @@ <h2>The <dfn>NDEFRecordType</dfn> string</h2>
       <td><dfn>"`empty`"</dfn></td>
       <td><i>unused</i></td>
       <td><i>unused</i></td>
-      <td><a>Empty record</a></td>
+      <td><a>Empty record</a> (TNF=0)</td>
     </tr>
     <tr>
       <td><dfn>"`text`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`T`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`T`"</td>
     </tr>
     <tr>
       <td><dfn>"`url`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{DOMString}}</td>
-      <td><a>Well-known type</a> record with type "`U`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`U`"</td>
     </tr>
     <tr>
       <td><dfn>"`smart-poster`"</dfn></td>
       <td><i>unused</i></td>
       <td>{{NDEFMessage}}</td>
-      <td><a>Well-known type</a> record with type "`Sp`"</td>
+      <td><a>Well-known type</a> (TNF=1) record with TYPE="`Sp`"</td>
+    </tr>
+    <tr>
+      <td><dfn>"`absolute-url`"</dfn></td>
+      <td><i>unused</i></td>
+      <td>{{DOMString}} <i>url</i></td>
+      <td><a>Absolute-URL record</a> (TNF=2) with TYPE=<i>url</i></td>
     </tr>
     <tr>
       <td><dfn>"`json`"</dfn></td>
       <td><a>JSON MIME type</a></td>
       <td>
         [= JSON type =]
       </td>
-      <td><a>MIME type record</a> with type equal to
-        <a>MIME type</a>.
+      <td><a>MIME type record</a> (TNF=2) with TYPE= [= JSON type =].
       </td>
     </tr>
     <tr>
       <td><dfn>"`opaque`"</dfn></td>
-      <td><a>MIME type</a></td>
+      <td>[= MIME type =]</a></td>
       <td>{{ArrayBuffer}}</td>
-      <td><a>MIME type record</a></td>
+      <td><a>MIME type record</a> (TNF=3) with TYPE= [= MIME type =]</td>
     </tr>
     <tr>
-      <td><a>external type</a></td>
+      <td><i>[= external type =]</i></td>
+      <td><i>unused</i></td>
+      <td>
+        {{ArrayBuffer}} or any<br>
+        <a>typed array type</a> or<br>
+        {{NDEFMessage}}
+      </td>
+      <td><a>External type record</a> (TNF=4) with TYPE= [= external type =]</td>
+    </tr>
+    <tr>
+      <td><i>[= local type =]</i></td>
       <td><i>unused</i></td>
       <td>
-        {{ArrayBuffer}} or<br>
```

Comment:
Why does a local type record data could only be ArrayBuffer or typed array? Given that this example  https://w3c.github.io/web-nfc/#push-a-smart-poster-message is feeding it with Number or Text.

Reply:
Local type can be anything an external type can be. They are basically external types (with different type names) that are scoped to an external record

Reply:
We need to add some prefix to these types - I have filed an issue about that today.

Reply:
No, we should not prefix. Local types should be only used inside a local context, the most well defined one being a parent record. At the moment, only wkt can have local types, but they could be also used by other types that allow embedded records, such as external type.

Reply:
I dont mean prefix when written to the record, but we need a way to differentiate

Reply:
The spec doesn't need to differentiate, apps do.

Reply:
I disagree but I don't want to block this PR on this, we can handle later

Reply:
It seems to me that you're saying a local type record is definitely an external record, right? But the record type being `text` conflicts with the WKT text record.

> {
recordType: "text",
data: "Funny dance"
},

Reply:
Local type records can be in WKT (like in "Sp" and handover), these are standardized.
They can be reused in other places where an application context exists, even with changed semantics.
In addition, new local type records can be used in application context, of course they should be chosen to not conflict (locally) with other types.
The term "application context" is not well defined in NFC. So far (by the standardized examples) it seems to mean the context is a parent record.

That parent record (context) that applications can actually use is currently only the external type records.

That is why IMHO prefixing is not needed, since the context already guarantees (local) uniqueness.

If the use case is to override a WKT with a local type of same name, then that is application territory and Web NFC should not add special smartness beyond the NFC specs. 
I would not like to instill and algorithm that would say 

> "if you see the 'local:' prefix in a type name, create an embedding external type record and add this record to an NDEFMessage that is set as payload to the embedding record".

I suggest we keep it simple for now.

That "text" example above is a text record, not a local type record. In other words, for "text" one doesn't need to define a local type. If encoding is special, use a MIME type or a plain external record.